Adapt lss handling

phettberg opened the pull request:
## Fixes (to be discussed):
Do not add the NMT reset to the scheduler after calling the `activate_bit_timing` command. When changing the bitrate on an OS (e.g., using `socketcan`, we have to reinitialize everything and thus stop the scheduler.

## Features
Provide the possibility to set a store configuration callback, to store newly set bitrate and node ID persistently. If it's not provided, the command still replies with the not supported value.

## Remarks
I'll add additional LSS test if we agree to a final solution

phettberg committed:
adapt LSS handler
* change baudrate configuration CB handling
* change pending baudrate from index to bitrate
* add store configuration functionality

phettberg committed:
update changelog

semiversus commented on CHANGELOG.md:
```diff
@@ -1,3 +1,8 @@
+# 0.4.13
```
I would go for 0.5.0

phettberg replied:
Done

semiversus commented on src/durand/services/lss.py:
```diff
@@ -18,24 +18,40 @@ class LSSState(IntEnum):
 
 
 class LSSSlave:
+    CiA_bit_timing_table = {  # in kBit
```
May store the values in Bit/s, so you don't have to recalculate later

phettberg replied:
Done

semiversus reviewed (commented):
This approach looks good.

phettberg committed:
avoid recalculating the bitrate

phettberg committed:
add docstrings for lss callbacks

phettberg committed:
semver minor increase

phettberg committed:
use a fixed image version for the test stage

phettberg committed:
add node id and baudrate configuration test

phettberg committed:
removed type hint for old python versions to work

phettberg commented:
* Changed test stage image to ubuntu-22.04 to remain compatible with Python 3.7 test.
* Added unittest for LSS configuration (node id, baudrate, store and activate bit timing)
* Added docstrings for the LSS callback function setters

phettberg commented:
*Be aware:* The activate bit timing command will *NOT* reset the NMT to initialize anymore, as its callback is responsible to reinitialize the network. Then, the NMT will start in initializing state anyway. If you still want the possibility to reset the NMT, we have to pass the NMT object to the callback as well. 

What's your opinion on this  @semiversus ?

phettberg commented:
Could you please review and merge @semiversus ?

Thank you very much!

semiversus commented:
Passing the NMT object to the callback is not necessary - as you wrote the stack will be reinitialized and NMT will start in initializing state. Pull request looks good.
